dbms/src/Storages/StorageDisaggregated.cpp

```diff
@@ -0,0 +1,334 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Coprocessor/InterpreterUtils.h>
+#include <Storages/StorageDisaggregated.h>
+#include <Storages/Transaction/TMTContext.h>
+
+namespace DB
+{
+const String StorageDisaggregated::ExecIDPrefixForTiFlashStorageSender = "exec_id_disaggregated_tiflash_storage_sender";
+
+BlockInputStreams StorageDisaggregated::read(
+    const Names &,
+    const SelectQueryInfo &,
+    const Context &,
+    QueryProcessingStage::Enum &,
+    size_t,
+    unsigned num_streams)
+{
+    auto remote_requests = buildRemoteRequests();
+    auto dispatch_reqs = buildAndDispatchMPPTaskRequests(remote_requests);
+
+    DAGPipeline pipeline;
```

Comment:
If error happens here, there is no chance to cancel the mpp task in storage node?(maybe not a big problem, since the mpp task in storage node will meet write timeout error)

Reply:
yeah...

Reply:
Move dispatch within Construction of `ExchangeReceiver`, so if anything failure happens between dispatch and ExchangeReceiver built done, we can cancel MPPTask.